### ISSUE TITLE
Verify logging and stdout/stderr scenarios in IIS/IISExpress

### ISSUE THREAD

**Issue by jkotalik** [aspnet/IISIntegration#1039]
This has bitten us a few times already. Diagnosing startup errors one of the main issues people run into using ANCM. Here is a list of all scenarios covered by redirecting stdout and stderr to logs and event logs.

### Logging to file
- Redirection starts once inprocess application begins. 
- Redirection ends once inprocess application ends.
- Will be logged to the file specified in web.config `stdoutLogFile`
- Redirects both stdout and stderr
- Primarily used for debugging applications

Scenarios to test:
- Run app specifying folder path, verify app output is stored in log file 
- Run app specifying folder path with invalid permissions, verify app output isn't written but app doesn't fail to start.

### Logging to pipe
- Redirection starts once inprocess application begins. 
- Redirection ends once inprocess application startup is complete.
- Added when `stdoutLogging=false` and there isn't an active console (for iis express scenarios)
- Creates an anonymous pipe to redirect console output to a local buffer, capturing startup exceptions and failures.
- Only captures first 4KB of output from managed
- After redirection, all output captured in the buffer is then logged to the original stdout source.
- Only IISExpress will have stdout/stderr before redirection, IIS doesn't set stdout and stderr on w3wp.exe

### Debug log file for ANCM
- Logs captured for ANCM shim and request handler
- Used by IIS and IISExpress for debugging ANCM issues.
- Enabled in either the web.config or environment variable (preferring web.config)

Scenarios to test:
- Debug logs present after enabling either source

**Comment by muratg** [comment]
Docs page: https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/azure-iis-errors-reference?view=aspnetcore-2.1

**Comment by jkotalik** [comment]
Closing in favor of https://github.com/aspnet/IISIntegration/issues/1448
